dm/syncer/dml_worker.go

```diff
@@ -312,3 +320,15 @@ func (w *DMLWorker) genSQLs(jobs []*job) ([]string, [][]interface{}) {
 	}
 	return queries, args
 }
+
+func (w *DMLWorker) judgeKeyNotFound(affect int, jobs []*job) bool {
+	if w.compact || w.multipleRows {
```

Comment:
for multipleRows, we can count the row numbers rather than job number.

And compact should also be OK?

Reply:
in multipleRows, update will be transform to insert on duplicate update, if downstream may lack the previous key and return wrong affect.

Reply:
for example
```
update 1,'aa'
update 1,'aaa'
```
will be compacted to one job and executed as followed
```
insert into xx(1,'aaa') on duplicate key update x=VALUES(x)
```
If downstream lack key 1, above sql will return affect=1. But acually it should return an error.

Reply:
for example
```
insert 1
delete 1
```
will be compacted to one job and executed as followed
```
delete 1
```
which will return affect=0, but it should not be an error.

Reply:
for example
```
delete 1
insert 1
```
will be compacted to one job and executed as followed
```
insert into xx(1) on duplicate key update x=VALUES(x)
```
If downstream lack key 1, above sql will return affect=1. But acually it should return an error when `delete 1` which has been compacted.

Reply:
Thx for your explanation.

Is this feature generally unavailable when users enable `compact` or `multipleRows`? In the situation demonstrated above, does DM just simply skip checking `affect`?

Reply:
Yes, because we now it's a warning log and we don't want to add a new config item, such as 'ErrKeyNotFound', so the `compact` and `mutipleRows` has higher priority.

Reply:
1. For multipleRows, we should be able to check the affected rows number is
   - len(job) for INSERT
   - 2*len(job) for INSERT ON DUPLIDATE KEY UDPATE. ref https://dev.mysql.com/doc/refman/5.6/en/insert-on-duplicate.html
   - len(job) for DELETE

2. For compacted job, we still can check that the compected DML should have expected affected rows number? DELETE + INSERT = UPDATE, and check it by the expected value of UPDATE or INSERT ON DUPLIDATE KEY UDPATE

Reply:
Compact: there are still many cases, like update primary key will be transform to DELETE+INSERT, which actually should be 1 row affected
MultipleRows: we need to calculate expected affected rows base on the sql type. I leave a TODO😂